### PR TITLE
Replacing Gerrit with GitLab

### PR DESCRIPTION
**Pull request by lykabernardo**

### CODE REVIEW

**Comment by davidjeddy** [comment]
Looks reasonable, Travis not being able to build the project indicates a configuration error though. Could we get that corrected please.

As a side note; I am personally not a fan of force-pushes. It re-writes the branch history on the repo. This makes it impossible for anyone else to contribute during the dev. cycle.

File: .travis.yml [hunk 1]
```diff
@@ -28,7 +28,9 @@ before_install:
 
 script:
  - ./adop
- - travis_wait 40 ./adop compose init
+ - while sleep 9m; do echo "=====[ $SECONDS seconds still running ]====="; done &
```
Comment (davidjeddy) [inline .travis.yml]: 9 minute wait? Can we do a loop every 10 seconds and exit when the service is ready? A flat 9m could be to short, could be way to long.

File: cmd/project [hunk 2]
```diff
@@ -49,7 +49,7 @@ load_help() {
     printf "    %-3s   %s\n" "" "-r <ENABLE_CODE_REVIEW>  : This is to control whether to enable code review or not for the repositories loaded by cartridge. (Optional) (Default: false)"
     printf "    %-3s   %s\n" "" "-h                   : Print this help."
     echo
-    echo "HINT: Run '${CMD_NAME} ${SUB_CMD_NAME} -p <PROJECT_NAME> -w <WORKSPACE_NAME> load -u ssh://jenkins@gerrit:29418/cartridges/adop-cartridge-java.git' to load java reference catridge in adop jenkins instance under existing workspace/project."
+    echo "HINT: Run '${CMD_NAME} ${SUB_CMD_NAME} -p <PROJECT_NAME> -w <WORKSPACE_NAME> load -u https://github.com/Accenture/adop-cartridge-java.git' to load java reference catridge in adop jenkins instance under existing workspace/project."
```
Comment (davidjeddy) [inline cmd/project]: Any way we can continue to use SSH w/ GitHub?

File: site/_docs/developing/cartridges.md [hunk 3]
```diff
@@ -31,13 +31,13 @@ High-level guidelines for writing Jenkins jobs in Job DSL:
     * Ensure that you are using the preBuildCleanup() wrapper in your Job DSL to make sure that your Jenkins workspace is cleared before every build.
     * Do not make any credentials visible anywhere in your job i.e. do not echo passwords in your shell script or store credentials as plain-text build parameters. Ensure you are always using the maskPasswords() and injectPasswords() wrappers in your Job DSL.
         * Credentials can also be added manually to the Jenkins credential store and then referenced in your job.
-    * It is highly recommended that the sshAgent() wrapper be used with the Jenkins credentials specified in order to allow Jenkins to clone all local git repositories from Gerrit over SSH.
+    * It is highly recommended that the sshAgent() wrapper be used with the Jenkins credentials specified in order to allow Jenkins to clone all local git repositories from Gitlab over SSH.
```
Comment (davidjeddy) [inline site/_docs/developing/cartridges.md]: Above in cmd/project HTTPS is used to clone.

File: site/_docs/operating/cartridges.md [hunk 4]
```diff
@@ -18,6 +18,6 @@ Once you [generate a project](/adop-docker-compose/docs/operating/projects), you
 	- `CARTRIDGE_FOLDER` - The folder name within the project namespace where your cartridge will be loaded into.
 	- `FOLDER_DISPLAY_NAME` - Display name of the folder where the cartridge is loaded.
 	- `FOLDER_DESCRIPTION` - Description of the folder where the cartridge is loaded.
-	- `ENABLE_CODE_REVIEW` - Enables Gerrit Code Reviewing for the selected cartridge.
+	- `ENABLE_CODE_REVIEW` - Enables Code Reviewing for the selected cartridge (only supports Gerrit).
```
Comment (davidjeddy) [inline site/_docs/operating/cartridges.md]: Gitlab is being added as the code review option, no?

File: site/_docs/reference/cli/project.md [hunk 5]
```diff
@@ -122,4 +122,4 @@ Used for creating a project space within a workspace in ADOP Jenkins instance.
 ## Examples [WIP]
 
 * `./adop project -p <PROJECT_NAME> -w <WORKSPACE_NAME> create` - creates a project in an existing workspace in ADOP Jenkins instance.
-* `./adop project -p <PROJECT_NAME> -w <WORKSPACE_NAME> load -u ssh://jenkins@gerrit:29418/cartridges/adop-cartridge-java.git` - loads Java reference catridge in ADOP Jenkins instance under existing workspace/project.
\ No newline at end of file
+* `./adop project -p <PROJECT_NAME> -w <WORKSPACE_NAME> load -u https://github.com/Accenture/adop-cartridge-java.git` - loads Java reference catridge in ADOP Jenkins instance under existing workspace/project.
```
Comment (davidjeddy) [inline site/_docs/reference/cli/project.md]: Samer comment as above regarding ssh vs https.

**Comment by davidjeddy** [comment]
I am still concerned the build is failing.

File: site/_docs/operating/cartridges.md (hunk 4 above)
Comment (davidjeddy) [reply site/_docs/operating/cartridges.md]: Ah, ok. That helps clarify it.

File: docker-compose.yml [hunk 6]
```diff
@@ -35,65 +35,81 @@ ldap:
...
+     external_url "${PROTO}://${TARGET_HOST}/gitlab"
+     postgresql['enable'] = false
+     gitlab_rails['db_username'] = "${GITLAB_POSTGRES_USER}"
+     gitlab_rails['db_password'] = "${GITLAB_POSTGRES_PASSWORD}"
+     gitlab_rails['db_host'] = "gitlab-postgresql"
+     gitlab_rails['db_port'] = "5432"
+     gitlab_rails['db_database'] = "${GITLAB_POSTGRES_DB}"
+     gitlab_rails['db_adapter'] = 'postgresql'
+     gitlab_rails['db_encoding'] = 'utf8'
+     gitlab_rails['monitoring_whitelist'] = ['127.0.0.0/8', '172.0.0.0/8']
```
Comment (RobertNorthard) [inline docker-compose.yml]: The docker network (local_network) is created with 172.19.0.0/16 address space, but this does not stop CUSTOM_NETWORK_NAME been overridden and a user using a pre-defined one.

I feel we have 3 options here:
* Whitelist the whole private IP address space and loopbacks
* Pass the docker network subnet CIDR in as a parameter
* Whitelist all and rely on the proxy / Docker SDN

File: env.config.sh [hunk 7]
```diff
@@ -17,22 +17,16 @@ export LDAP_GROUPMEMBERPATTERN='(uniqueMember=${dn})'
 export LDAP_MANAGER_DN="cn=admin,dc=ldap,dc=example,dc=com"
 export LDAP_GROUP_NAME_ADMIN="administrators"
 
-# Gerrit MySQL
+# Gitlab Postgresql
+export GITLAB_POSTGRES_USER="gitlab"
+export GITLAB_POSTGRES_PASSWORD="gitlab"
```
[the remark this answers is not in the document]
Comment (lykabernardo) [reply env.config.sh]: Updated PR to include generation or postgres password.

File: docker-compose.yml (hunk 6 above)
Comment (lykabernardo) [reply docker-compose.yml]: As discussed with @marisbahtins, updated PR with the second option: Pass the docker network subnet CIDR in as a parameter

File: .travis.yml (hunk 1 above)
Comment (davidjeddy) [reply .travis.yml]: That is an interesting problem.

 - It looks like TravisCI provisions 2cpu / 7.5 mem for linux build ENVs. (https://docs.travis-ci.com/user/reference/overview/)
 - GitLab looks to req. 1CPU+4Gb as a base level. (https://docs.gitlab.com/ce/install/requirements.html)
 - TravisCI also has a max. execution limit of around 30 minutes as per the build logs (https://travis-ci.org/Accenture/adop-docker-compose/builds/465007061?utm_source=github_status&utm_medium=notification). Ideally build/test steps should be < 10 minutes. I `feel` like a different resolution is needed here other than increasing the build timeout. Maybe use a pre-build TravisCI container image?

**Pull request closed without merging**